rust/otap-dataflow/crates/otap/src/retry_processor.rs

```diff
@@ -73,15 +33,150 @@ use otap_df_telemetry::instrument::Counter;
 use otap_df_telemetry::metrics::MetricSet;
 use otap_df_telemetry_macros::metric_set;
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
+use serde_with::{DurationSecondsWithFrac, formats::Flexible, serde_as};
 use std::sync::Arc;
-use std::time::{Duration, Instant};
-
-/// Maximum age for failed messages before cleanup (5 minutes)
-const MAX_FAILED_MESSAGE_AGE_SECS: u64 = 300;
+use std::time::{Duration, Instant, SystemTime};
 
 /// URN for the RetryProcessor processor
-pub const RETRY_PROCESSOR_URN: &str = "urn:otap:processor:retry_processor";
+pub const RETRY_PROCESSOR_URN: &str = "urn:otel:retry:processor";
+
+/// Configuration for the retry processor. Modeled on
+/// https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md#retry-on-failure.
+#[serde_as]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct RetryConfig {
```

Comment:
Not for this PR, but eventually, shouldn't we introduce a jitter mechanism?

Reply:
Yes. Added a ToDo.